lib/puppet/type/file/source.rb

```diff
@@ -220,11 +220,11 @@ def full_path
     end
 
     def server?
-       uri and uri.host
+       uri && uri.host && uri.host.length > 0
```

Comment:
Why not `.empty?` here?

Reply:
No particular reason, puppet does both, though it looks like `empty?` is the preferred option. I'll update that.